### PR TITLE
[#515] Replace hedgehog state machine model with Trace-based one

### LINKED ISSUE #515: Incorporate New UTxO Generators in UTXOW State Machine
## Background

The initial generators that we used for the UTXOW state machine testing relied heavily on the `Monad` instance of `Gen` and therefore have relatively poor shrinking properties. input-output-hk/cardano-ledger-specs#489 replaces these generators to use the applicative instance and manual shrinking. We should update the state machine model to use these new generators.

## Proposed Solution

Update the state machine generators to use the new ones from `cardano-ledger-specs`.

## Next Action
- [ ] Update the state machine generators to use the new ones from `cardano-ledger-specs`
- [ ] Make sure that the UTXOW model tests still pass

### PR DESCRIPTION
**Pull request by ruhatch**
Closes #515

### CODE REVIEW

File: test/Test/Cardano/Chain/Block/Validation/Spec.hs [hunk 1]
```diff
@@ -51,7 +52,7 @@ tests = $$discoverPropArg
 ts_prop_generatedChainsAreValidated :: TSProperty
 ts_prop_generatedChainsAreValidated =
   withTestsTS 300 $ property $ do
-    tr <- forAll trace
+    tr <- forAll $ trace @CHAIN 100
```
Comment (dnadales) [inline test/Test/Cardano/Chain/Block/Validation/Spec.hs:55]: Would it make more sense to test `100` times with traces of length `300`?

What happens if we extend this to `200`?

File: test/Test/Cardano/Chain/UTxO/Model.hs [hunk 2]
```diff
@@ -1,303 +1,118 @@
...
-{-# LANGUAGE RankNTypes          #-}
-{-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE TemplateHaskell     #-}
-{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TupleSections     #-}
+{-# LANGUAGE TypeApplications  #-}
 
-{-# OPTIONS_GHC -fno-warn-orphans #-}
```
Comment (dnadales) [inline test/Test/Cardano/Chain/UTxO/Model.hs:13]: :tada:

**Review by dnadales — APPROVED** [review]
Looks great! Let's see how far traces can get us :four_leaf_clover:

File: test/Test/Cardano/Chain/Block/Validation/Spec.hs (hunk 1 above)
Comment (ruhatch) [reply test/Test/Cardano/Chain/Block/Validation/Spec.hs:55]: Not sure yet, as this one's not hooked up :P I'm going to revive it in the next PR

Comment (ruhatch) [reply test/Test/Cardano/Chain/Block/Validation/Spec.hs:55]: But yeah, not sure what the best parameters are for these

**Pull request merged**